madmom/features/beats.py

```diff
@@ -554,6 +569,72 @@ def process(self, activations, **kwargs):
         # remove beats with negative times and return them
         return detections[np.searchsorted(detections, 0):]
 
+    def process_online(self, activations, reset=True, **kwargs):
+        """
+        Detect the beats in the given activation function for online mode.
+
+        Parameters
+        ----------
+        activations : numpy array
+            Beat activation function.
+        reset : bool, optional
+            Reset the BeatTrackingProcessor to its initial state before
+            processing.
+
+        Returns
+        -------
+        beats : numpy array
+            Detected beat positions [seconds].
+
+        """
+        # reset to initial state
+        if reset:
+            self.reset()
+        beats_ = []
+        for activation in activations:
+            # shift buffer and put new activation at end of buffer
+            buffer = self.buffer(activation)
+            # create an online interval histogram
+            histogram = self.tempo_estimator.online_interval_histogram(
+                activation, reset=False)
+            # get the dominant interval
+            interval = self.tempo_estimator.dominant_interval(histogram)
+            # compute current and next beat times
+            cur_beat = self.counter / float(self.fps)
+            next_beat = self.last_beat + 60. / self.tempo_estimator.max_bpm
+            # only detect beats again after at least min_interval frames
+            detections = []
+            if cur_beat >= next_beat:
+                detections = detect_beats(buffer, interval, self.look_aside)
+            # if a detection falls within the last few frames and exceeds a
+            # certain threshold a beat was detected. this is important because
+            # for every frame the tempo may change and therefore the last beat
+            # can easily be missed.
+            if len(detections) > 0 and detections[-1] >= len(buffer) - 5 and \
+               buffer[detections[-1]] > self.threshold:
```

Comment:
Ah ok, cool! Yeah using only "active" segments seems like a good idea. Maybe the distance between those active segments gives a hint whether it should stop tracking or not. 
For online it's a bit different anyway because we don't know how long the break will be so we have to decide on the spot (or maybe with a bit of delay).